src/lib/index.js

```diff
@@ -474,7 +474,7 @@ lib.mergeArray = function(traceAttr, cd, cdAttr, fn) {
 lib.mergeArrayCastPositive = function(traceAttr, cd, cdAttr) {
     return lib.mergeArray(traceAttr, cd, cdAttr, function(v) {
         var w = +v;
-        return w > 0 ? w : 0;
+        return isNaN(w) ? NaN : w > 0 ? w : 0;
```

Comment:
I think you want `isNumeric` here. Have tried something like

```js
'marker.line.width': [NaN, null, undefined, [], {}, 0]`
```

?

Reply:
Works OK with those cases. Using `isNumeric` might be slow.

Reply:
Have you tried running benchmarks here? I doubt that `isNumeric` is that much slower than `isNaN`

Reply:
Moreover, could we lock those cases down in a `_module.calc` test?

Reply:
Tests are added in https://github.com/plotly/plotly.js/pull/4063/commits/39b9840933957640fe26e9a9f9c30e1a5625c782.